lib/graphql_metrics/analyzer.rb

```diff
@@ -0,0 +1,193 @@
+module GraphQLMetrics
+  class Analyzer < GraphQL::Analysis::AST::Analyzer
```

Comment:
It feels like there are two different operations happening inside of this class:
1. The analyzer is adding callbacks for metrics per field
2. There is a setup beforehand that creates the callbacks etc. Should these two concepts be split into different classes?